File: DXDAL/Receiving.vb
Imports System
Imports DevExpress.Xpo
Imports DevExpress.Data.Filtering
Imports DevExpress.Xpo.Helpers

Namespace SPGData
    <Persistent("tblReceivMain")> _
    Public Class Receiving
        Inherits XPLiteObject
        Dim fReceivID As Integer
        <Key()> _
        Public Property ReceivID() As Integer
            Get
                Return fReceivID
            End Get
            Set(ByVal value As Integer)
                SetPropertyValue(Of Integer)("ReceivID", fReceivID, value)
            End Set
        End Property
        Dim fReceivDate As DateTime
        Public Property ReceivDate() As DateTime
            Get
                Return fReceivDate
            End Get
            Set(ByVal value As DateTime)
                SetPropertyValue(Of DateTime)("ReceivDate", fReceivDate, value)
            End Set
        End Property
        Dim fReceivCarrierID As Carriers
        Public Property ReceivCarrierID() As Carriers
            Get
                Return fReceivCarrierID
            End Get
            Set(ByVal value As Carriers)
                SetPropertyValue(Of Carriers)("ReceivCarrierID", fReceivCarrierID, value)
            End Set
        End Property
        Dim fReceivCustID As Customers
        Public Property ReceivCustID() As Customers
            Get
                Return fReceivCustID
            End Get
            Set(ByVal value As Customers)
                SetPropertyValue(Of Customers)("ReceivCustID", fReceivCustID, value)
            End Set
        End Property
        Dim fReceivBOL As String
        <Size(50)> _
        Public Property ReceivBOL() As String
            Get
                Return fReceivBOL
            End Get
            Set(ByVal value As String)
                SetPropertyValue(Of String)("ReceivBOL", fReceivBOL, value)
            End Set
        End Property
        Dim fstrEnteredBy As String
        <Size(50)> _
        Public Property strEnteredBy() As String
            Get
                Return fstrEnteredBy
            End Get
            Set(ByVal value As String)
                SetPropertyValue(Of String)("strEnteredBy", fstrEnteredBy, value)
            End Set
        End Property
        Dim fdtmEnteredOn As DateTime
        Public Property dtmEnteredOn() As DateTime
            Get
                Return fdtmEnteredOn
            End Get
            Set(ByVal value As DateTime)
                SetPropertyValue(Of DateTime)("dtmEnteredOn", fdtmEnteredOn, value)
            End Set
        End Property
        Dim fstrVendor As Vendor
        <Association("Receiving-Vendor")> _
        Public Property strVendor() As vendor
            Get
                Return fstrVendor
            End Get
            Set(ByVal value As Vendor)
                SetPropertyValue(Of Vendor)("strVendor", fstrVendor, value)
            End Set
        End Property
        Dim fsngSkits As Single
        Public Property sngSkits() As Single
            Get
                Return fsngSkits
            End Get
            Set(ByVal value As Single)
                SetPropertyValue(Of Single)("sngSkits", fsngSkits, value)
            End Set
        End Property
        Dim fsngTotalPallets As Single
        Public Property sngTotalPallets() As Single
            Get
                Return fsngTotalPallets
            End Get
            Set(ByVal value As Single)
                SetPropertyValue(Of Single)("sngTotalPallets", fsngTotalPallets, value)
            End Set
        End Property
        Dim fstrPO As String
        <Size(50)> _
        Public Property strPO() As String
            Get
                Return fstrPO
            End Get
            Set(ByVal value As String)
                SetPropertyValue(Of String)("strPO", fstrPO, value)
            End Set
        End Property
        Dim fintShiftID As Shifts
        Public Property intShiftID() As Shifts
            Get
                Return fintShiftID
            End Get
            Set(ByVal value As Shifts)
                SetPropertyValue(Of Shifts)("intShiftID", fintShiftID, value)
            End Set
        End Property
        Dim fstrTrailer As String
        <Size(50)> _
        Public Property strTrailer() As String
            Get
                Return fstrTrailer
            End Get
            Set(ByVal value As String)
                SetPropertyValue(Of String)("strTrailer", fstrTrailer, value)
            End Set
        End Property
        Dim fstrSeal As String
        <Size(50)> _
        Public Property strSeal() As String
            Get
                Return fstrSeal
            End Get
            Set(ByVal value As String)
                SetPropertyValue(Of String)("strSeal", fstrSeal, value)
            End Set
        End Property
        Dim fNotes As String
        <Size(255)> _
        Public Property Notes() As String
            Get
                Return fNotes
            End Get
            Set(ByVal value As String)
                SetPropertyValue(Of String)("Notes", fNotes, value)
            End Set
        End Property
        Dim fUnloadedBy As Employees
        <Association("Employee-Unloaded")>
        Public Property UnloadedBy() As Employees
            Get
                Return fUnloadedBy
            End Get
            Set(value As Employees)
                SetPropertyValue(Of Employees)("UnloadedBy", fUnloadedBy, value)
            End Set
        End Property
        Dim fCheckedBy As Employees
        <Association("Employee-CheckedReceiving")>
        Public Property CheckedBy() As Employees
            Get
                Return fCheckedBy
            End Get
            Set(value As Employees)
                SetPropertyValue(Of Employees)("CheckedBy", fCheckedBy, value)
            End Set
        End Property
        Dim fPhysicalCondition As ShipmentConditions
        <Association("ReceivingPhysicalCondition")>
        Public Property PhysicalCondition As ShipmentConditions
            Get
                Return fPhysicalCondition
            End Get
            Set(value As ShipmentConditions)
                SetPropertyValue(Of ShipmentConditions)("PhysicalCondition", fPhysicalCondition, value)
            End Set
        End Property
        Dim fLoadCondition As ShipmentConditions
        <Association("ReceivingLoadCondition")>
        Public Property LoadCondition As ShipmentConditions
            Get
                Return fLoadCondition
            End Get
            Set(value As ShipmentConditions)
                SetPropertyValue(Of ShipmentConditions)("LoadCondition", fLoadCondition, value)
            End Set
        End Property
        Dim fStartTime As DateTime
        Public Property StartTime As DateTime
            Get
                Return fStartTime
            End Get
            Set(value As DateTime)
                SetPropertyValue(Of DateTime)("StartTime", fStartTime, value)
            End Set
        End Property
        Dim fFinishTime As DateTime
        Public Property FinishTime As DateTime
            Get
                Return fFinishTime
            End Get
            Set(value As DateTime)
                SetPropertyValue(Of DateTime)("FinishTime", fFinishTime, value)
            End Set
        End Property
        Dim fTemparature As Double
        Public Property Temparature As Double
            Get
                Return fTemparature
            End Get
            Set(value As Double)
                SetPropertyValue(Of Double)("Temparature", fTemparature, value)
            End Set
        End Property
        Dim fWheelsChocked As ThreeWayYesNo
        Public Property WheelsChocked As ThreeWayYesNo
            Get
                Return fWheelsChocked
            End Get
            Set(value As ThreeWayYesNo)
                SetPropertyValue(Of ThreeWayYesNo)("WheelsChocked", fWheelsChocked, value)
            End Set
        End Property
        Dim fForeignSubstance As ThreeWayYesNo
        Public Property ForeignSubstance As ThreeWayYesNo
            Get
                Return fForeignSubstance
            End Get
            Set(value As ThreeWayYesNo)
                SetPropertyValue(Of ThreeWayYesNo)("ForeignSubstance", fForeignSubstance, value)
            End Set
        End Property
        Dim fInsectActivity As ThreeWayYesNo
        Public Property InsectActivity As ThreeWayYesNo
            Get
                Return fInsectActivity
            End Get
            Set(value As ThreeWayYesNo)
                SetPropertyValue(Of ThreeWayYesNo)("InsectActivity", fInsectActivity, value)
            End Set
        End Property
        Dim fCorrectPallets As ThreeWayYesNo
        Public Property CorrectPallets As ThreeWayYesNo
            Get
                Return fCorrectPallets
            End Get
            Set(value As ThreeWayYesNo)
                SetPropertyValue(Of ThreeWayYesNo)("CorrectPallets", fCorrectPallets, value)
            End Set
        End Property
        Dim fReceivingLocation As Locations
        <Association("Location-Receivings")>
        Public Property ReceivingLocation As Locations
            Get
                Return fReceivingLocation
            End Get
            Set(value As Locations)
                SetPropertyValue(Of Locations)("ReceivingLocation", fReceivingLocation, value)
            End Set
        End Property

        <Association("Receiving-Details", GetType(ReceivingDetail)), Aggregated()> _
        Public ReadOnly Property ReceivingDetails() As XPCollection(Of ReceivingDetail)
            Get
                Return GetCollection(Of ReceivingDetail)("ReceivingDetails")
            End Get
        End Property
        <Association("Receiving-Returns", GetType(ReceivedReturns)), Aggregated()> _
        Public ReadOnly Property ReceivingReturns() As XPCollection(Of ReceivedReturns)
            Get
                Return GetCollection(Of ReceivedReturns)("ReceivingReturns")
            End Get
        End Property
        Public Sub New(ByVal session As Session)
            MyBase.New(session)
        End Sub
        Public Sub New()
            MyBase.New(Session.DefaultSession)
        End Sub
        Public Overrides Sub AfterConstruction()
            MyBase.AfterConstruction()
        End Sub
#Region "XPO nested fields class - don't edit manually"
        Public Shadows Class FieldsClass
            Inherits DevExpress.Xpo.PersistentBase.FieldsClass
            Public Sub New()
                MyBase.New()
            End Sub
            Public Sub New(ByVal propertyName As String)
                MyBase.New(propertyName)
            End Sub
            Public ReadOnly Property ReceivID() As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("ReceivID"))
                End Get
            End Property
            Public ReadOnly Property ReceivDate() As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("ReceivDate"))
                End Get
            End Property
            Public ReadOnly Property ReceivCarrierID() As Carriers.FieldsClass
                Get
                    Return New Carriers.FieldsClass(GetNestedName("ReceivCarrierID"))
                End Get
            End Property
            Public ReadOnly Property ReceivCustID() As Customers.FieldsClass
                Get
                    Return New Customers.FieldsClass(GetNestedName("ReceivCustID"))
                End Get
            End Property
            Public ReadOnly Property ReceivBOL() As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("ReceivBOL"))
                End Get
            End Property
            Public ReadOnly Property strEnteredBy() As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("strEnteredBy"))
                End Get
            End Property
            Public ReadOnly Property dtmEnteredOn() As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("dtmEnteredOn"))
                End Get
            End Property
            Public ReadOnly Property strVendor() As Vendor.FieldsClass
                Get
                    Return New vendor.FieldsClass(GetNestedName("strVendor"))
                End Get
            End Property
            Public ReadOnly Property sngSkits() As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("sngSkits"))
                End Get
            End Property
            Public ReadOnly Property sngTotalPallets() As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("sngTotalPallets"))
                End Get
            End Property
            Public ReadOnly Property strPO() As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("strPO"))
                End Get
            End Property
            Public ReadOnly Property intShiftID() As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("intShiftID"))
                End Get
            End Property
            Public ReadOnly Property strTrailer() As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("strTrailer"))
                End Get
            End Property
            Public ReadOnly Property strSeal() As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("strSeal"))
                End Get
            End Property
            Public ReadOnly Property Notes() As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("Notes"))
                End Get
            End Property
            Public ReadOnly Property UnloadedBy() As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("UnloadedBy"))
                End Get
            End Property
            Public ReadOnly Property CheckedBy() As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("CheckedBy"))
                End Get
            End Property
            Public ReadOnly Property PhysicalCondition As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("PhysicalCondition"))
                End Get
            End Property
            Public ReadOnly Property LoadCondition As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("LoadCondition"))
                End Get
            End Property
            Public ReadOnly Property StartTime As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("StartTime"))
                End Get
            End Property
            Public ReadOnly Property FinishTime As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("FinishTime"))
                End Get
            End Property
            Public ReadOnly Property Temparature As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("Temparature"))
                End Get
            End Property
            Public ReadOnly Property WheelsChocked As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("WheelsChocked"))
                End Get
            End Property
            Public ReadOnly Property ForeignSubstance As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("ForeignSubstance"))
                End Get
            End Property
            Public ReadOnly Property InsectActivity As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("InsectActivity"))
                End Get
            End Property
            Public ReadOnly Property CorrectPallets As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("CorrectPallets"))
                End Get
            End Property
            Public ReadOnly Property ReceivingLocation As Locations.FieldsClass
                Get
                    Return New Locations.FieldsClass(GetNestedName("ReceivingLocation"))
                End Get
            End Property
        End Class
        Private Shared _fields As FieldsClass
        Public Shared Shadows ReadOnly Property Fields() As FieldsClass
            Get
                If ReferenceEquals(_fields, Nothing) Then
                    _fields = New FieldsClass()
                End If
                Return _fields
            End Get
        End Property
#End Region
    End Class
End Namespace

File: DXDAL/Receiving.cs
using System.Data;
using System.Diagnostics;
using Microsoft.VisualBasic;
using System.Collections.Generic;
using System.Collections;
using System;
using DevExpress.Xpo;
using DevExpress.Data.Filtering;
using DevExpress.Xpo.Helpers;

namespace DXDAL
{
    namespace SPGData
    {
        [Persistent("tblReceivMain")]
        public class Receiving : XPLiteObject
        {
            private int fReceivID;
            [Key()]
            public int ReceivID
            {
                get => fReceivID;
                set => SetPropertyValue<int>("ReceivID", ref fReceivID, value);
            }
            private DateTime fReceivDate;
            public DateTime ReceivDate
            {
                get => fReceivDate;
                set => SetPropertyValue<DateTime>("ReceivDate", ref fReceivDate, value);
            }
            private Carriers fReceivCarrierID;
            public Carriers ReceivCarrierID
            {
                get => fReceivCarrierID;
                set => SetPropertyValue("ReceivCarrierID", ref fReceivCarrierID, value);
            }
            private Customers fReceivCustID;
            public Customers ReceivCustID
            {
                get => fReceivCustID;
                set => SetPropertyValue("ReceivCustID", ref fReceivCustID, value);
            }
            private string fReceivBOL;
            [Size(50)]
            public string ReceivBOL
            {
                get => fReceivBOL;
                set => SetPropertyValue("ReceivBOL", ref fReceivBOL, value);
            }
            private string fstrEnteredBy;
            [Size(50)]
            public string strEnteredBy
            {
                get => fstrEnteredBy;
                set => SetPropertyValue("strEnteredBy", ref fstrEnteredBy, value);
            }
            private DateTime fdtmEnteredOn;
            public DateTime dtmEnteredOn
            {
                get => fdtmEnteredOn;
                set => SetPropertyValue<DateTime>("dtmEnteredOn", ref fdtmEnteredOn, value);
            }
            private Vendor fstrVendor;
            [Association("Receiving-Vendor")]
            public Vendor strVendor
            {
                get => fstrVendor;
                set => SetPropertyValue("strVendor", ref fstrVendor, value);
            }
            private float fsngSkits;
            public float sngSkits
            {
                get => fsngSkits;
                set => SetPropertyValue("sngSkits", ref fsngSkits, value);
            }
            private float fsngTotalPallets;
            public float sngTotalPallets
            {
                get => fsngTotalPallets;
                set => SetPropertyValue("sngTotalPallets", ref fsngTotalPallets, value);
            }
            private string fstrPO;
            [Size(50)]
            public string strPO
            {
                get => fstrPO;
                set => SetPropertyValue("strPO", ref fstrPO, value);
            }
            private Shifts fintShiftID;
            public Shifts intShiftID
            {
                get => fintShiftID;
                set => SetPropertyValue("intShiftID", ref fintShiftID, value);
            }
            private string fstrTrailer;
            [Size(50)]
            public string strTrailer
            {
                get => fstrTrailer;
                set => SetPropertyValue("strTrailer", ref fstrTrailer, value);
            }
            private string fstrSeal;
            [Size(50)]
            public string strSeal
            {
                get => fstrSeal;
                set => SetPropertyValue("strSeal", ref fstrSeal, value);
            }
            private string fNotes;
            [Size(255)]
            public string Notes
            {
                get => fNotes;
                set => SetPropertyValue("Notes", ref fNotes, value);
            }
            private Employees fUnloadedBy;
            [Association("Employee-Unloaded")]
            public Employees UnloadedBy
            {
                get => fUnloadedBy;
                set => SetPropertyValue("UnloadedBy", ref fUnloadedBy, value);
            }
            private Employees fCheckedBy;
            [Association("Employee-CheckedReceiving")]
            public Employees CheckedBy
            {
                get => fCheckedBy;
                set => SetPropertyValue("CheckedBy", ref fCheckedBy, value);
            }
            private ShipmentConditions fPhysicalCondition;
            [Association("ReceivingPhysicalCondition")]
            public ShipmentConditions PhysicalCondition
            {
                get => fPhysicalCondition;
                set => SetPropertyValue("PhysicalCondition", ref fPhysicalCondition, value);
            }
            private ShipmentConditions fLoadCondition;
            [Association("ReceivingLoadCondition")]
            public ShipmentConditions LoadCondition
            {
                get => fLoadCondition;
                set => SetPropertyValue("LoadCondition", ref fLoadCondition, value);
            }
            private DateTime fStartTime;
            public DateTime StartTime
            {
                get => fStartTime;
                set => SetPropertyValue<DateTime>("StartTime", ref fStartTime, value);
            }
            private DateTime fFinishTime;
            public DateTime FinishTime
            {
                get => fFinishTime;
                set => SetPropertyValue<DateTime>("FinishTime", ref fFinishTime, value);
            }
            private double fTemparature;
            public double Temparature
            {
                get => fTemparature;
                set => SetPropertyValue("Temparature", ref fTemparature, value);
            }
            private ThreeWayYesNo fWheelsChocked;
            public ThreeWayYesNo WheelsChocked
            {
                get => fWheelsChocked;
                set => SetPropertyValue("WheelsChocked", ref fWheelsChocked, value);
            }
            private ThreeWayYesNo fForeignSubstance;
            public ThreeWayYesNo ForeignSubstance
            {
                get => fForeignSubstance;
                set => SetPropertyValue("ForeignSubstance", ref fForeignSubstance, value);
            }
            private ThreeWayYesNo fInsectActivity;
            public ThreeWayYesNo InsectActivity
            {
                get => fInsectActivity;
                set => SetPropertyValue("InsectActivity", ref fInsectActivity, value);
            }
            private ThreeWayYesNo fCorrectPallets;
            public ThreeWayYesNo CorrectPallets
            {
                get => fCorrectPallets;
                set => SetPropertyValue("CorrectPallets", ref fCorrectPallets, value);
            }
            private Locations fReceivingLocation;
            [Association("Location-Receivings")]
            public Locations ReceivingLocation
            {
                get => fReceivingLocation;
                set => SetPropertyValue("ReceivingLocation", ref fReceivingLocation, value);
            }

            [Association("Receiving-Details", typeof(ReceivingDetail))]
            [Aggregated()]
            public XPCollection<ReceivingDetail> ReceivingDetails => GetCollection<ReceivingDetail>("ReceivingDetails");
            [Association("Receiving-Returns", typeof(ReceivedReturns))]
            [Aggregated()]
            public XPCollection<ReceivedReturns> ReceivingReturns => GetCollection<ReceivedReturns>("ReceivingReturns");
            public Receiving(Session session) : base(session)
            {
            }
            public Receiving() : base(Session.DefaultSession)
            {
            }
            public override void AfterConstruction()
            {
                base.AfterConstruction();
            }
            public new class FieldsClass : PersistentBase.FieldsClass
            {
                public FieldsClass() : base()
                {
                }
                public FieldsClass(string propertyName) : base(propertyName)
                {
                }
                public OperandProperty ReceivID => new OperandProperty(GetNestedName("ReceivID"));
                public OperandProperty ReceivDate => new OperandProperty(GetNestedName("ReceivDate"));
                public Carriers.FieldsClass ReceivCarrierID => new Carriers.FieldsClass(GetNestedName("ReceivCarrierID"));
                public Customers.FieldsClass ReceivCustID => new Customers.FieldsClass(GetNestedName("ReceivCustID"));
                public OperandProperty ReceivBOL => new OperandProperty(GetNestedName("ReceivBOL"));
                public OperandProperty strEnteredBy => new OperandProperty(GetNestedName("strEnteredBy"));
                public OperandProperty dtmEnteredOn => new OperandProperty(GetNestedName("dtmEnteredOn"));
                public Vendor.FieldsClass strVendor => new Vendor.FieldsClass(GetNestedName("strVendor"));
                public OperandProperty sngSkits => new OperandProperty(GetNestedName("sngSkits"));
                public OperandProperty sngTotalPallets => new OperandProperty(GetNestedName("sngTotalPallets"));
                public OperandProperty strPO => new OperandProperty(GetNestedName("strPO"));
                public OperandProperty intShiftID => new OperandProperty(GetNestedName("intShiftID"));
                public OperandProperty strTrailer => new OperandProperty(GetNestedName("strTrailer"));
                public OperandProperty strSeal => new OperandProperty(GetNestedName("strSeal"));
                public OperandProperty Notes => new OperandProperty(GetNestedName("Notes"));
                public OperandProperty UnloadedBy => new OperandProperty(GetNestedName("UnloadedBy"));
                public OperandProperty CheckedBy => new OperandProperty(GetNestedName("CheckedBy"));
                public OperandProperty PhysicalCondition => new OperandProperty(GetNestedName("PhysicalCondition"));
                public OperandProperty LoadCondition => new OperandProperty(GetNestedName("LoadCondition"));
                public OperandProperty StartTime => new OperandProperty(GetNestedName("StartTime"));
                public OperandProperty FinishTime => new OperandProperty(GetNestedName("FinishTime"));
                public OperandProperty Temparature => new OperandProperty(GetNestedName("Temparature"));
                public OperandProperty WheelsChocked => new OperandProperty(GetNestedName("WheelsChocked"));
                public OperandProperty ForeignSubstance => new OperandProperty(GetNestedName("ForeignSubstance"));
                public OperandProperty InsectActivity => new OperandProperty(GetNestedName("InsectActivity"));
                public OperandProperty CorrectPallets => new OperandProperty(GetNestedName("CorrectPallets"));
                public Locations.FieldsClass ReceivingLocation => new Locations.FieldsClass(GetNestedName("ReceivingLocation"));
            }
            private static FieldsClass _fields;
            public static new FieldsClass Fields
            {
                get
                {
                    if (ReferenceEquals(_fields, null))
                        _fields = new FieldsClass();
                    return _fields;
                }
            }
        }
    }
}
